apps/dashboard/src/components/workflow-editor/steps/controls/select-widget.tsx

```diff
@@ -24,7 +26,15 @@ export function SelectWidget(props: WidgetProps) {
         <FormItem className="my-2 py-1">
           <FormLabel>{capitalize(label)}</FormLabel>
           <FormControl>
-            <Select value={field.value} onValueChange={field.onChange} disabled={readonly} required={required}>
+            <Select
+              value={field.value}
+              onValueChange={(value) => {
+                field.onChange(value);
+                flushFormUpdates();
```

Comment:
All selects, dropdowns, toggles etc, basically not text fields should call `flushFormUpdates` after field changes.